test/e2e/upgrades/ingress.go

```diff
@@ -0,0 +1,113 @@
+
+	// gceController handles all cloud testing logic
+	gceController := &framework.GCEIngressController{
+		Ns:     ns.Name,
+		Client: jig.Client,
+		Cloud:  framework.TestContext.CloudConfig,
+	}
+
+	t.gceController = gceController
+	t.jig = jig
+	t.httpClient = framework.BuildInsecureClient(framework.IngressReqTimeout)
+
+	// Allocate a static-ip for the Ingress, this IP is cleaned up via CleanupGCE
+	t.ip = t.gceController.CreateStaticIP(ns.Name)
+
+	// Create a working basic Ingress
+	By(fmt.Sprintf("allocated static ip %v: %v through the GCE cloud provider", ns.Name, t.ip))
+	jig.CreateIngress(filepath.Join(framework.IngressManifestPath, "static-ip"), ns.Name, map[string]string{
+		"kubernetes.io/ingress.global-static-ip-name": ns.Name,
+		"kubernetes.io/ingress.allow-http":            "false",
+	})
+
+	By("waiting for Ingress to come up with ip: " + t.ip)
+	framework.ExpectNoError(framework.PollURL(fmt.Sprintf("https://%v/", t.ip), "", framework.LoadBalancerPollTimeout, jig.PollInterval, t.httpClient, false))
+}
+
+// Test waits for the upgrade to complete, and then verifies
+// with a connectvity check to the loadbalancer ip.
+func (t *IngressUpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade UpgradeType) {
+	switch upgrade {
+	case MasterUpgrade:
+		// Restarting the ingress controller shouldn't disrupt a steady state
+		// Ingress. Restarting the ingress controller and deleting ingresses
+		// while it's down will leak cloud resources, because the ingress
+		// controller doesn't checkpoint to disk.
+		t.verify(f, done, true)
+	default:
+		// Currently ingress gets disrupted across node upgrade, because endpoints
+		// get killed and we don't have any guarantees that 2 nodes don't overlap
+		// their upgrades (even on cloud platforms like GCE, because VM level
+		// rolling upgrades are not Kubernetes aware).
+		t.verify(f, done, false)
+	}
+}
+
+// Teardown cleans up any remaining resources.
+func (t *IngressUpgradeTest) Teardown(f *framework.Framework) {
+	if CurrentGinkgoTestDescription().Failed {
+		framework.DescribeIng(t.gceController.Ns)
+	}
+	if t.jig.Ingress == nil {
+		By("No ingress created, no cleanup necessary")
+		return
+	}
+	By("Deleting ingress")
+	t.jig.DeleteIngress()
+
+	By("Cleaning up cloud resources")
+	framework.CleanupGCE(t.gceController)
+}
+
+func (t *IngressUpgradeTest) verify(f *framework.Framework, done <-chan struct{}, testDuringDisruption bool) {
+	if testDuringDisruption {
+		By("continuously hitting the Ingress IP")
+		framework.ExpectNoError(framework.PollURL(fmt.Sprintf("https://%v/", t.ip), "", framework.LoadBalancerPollTimeout, t.jig.PollInterval, t.httpClient, false))
+	} else {
+		By("waiting for upgrade to finish without checking if Ingress remains up")
+		<-done
+	}
+	By("hitting the Ingress IP " + t.ip)
+	framework.ExpectNoError(framework.PollURL(fmt.Sprintf("https://%v/", t.ip), "", framework.LoadBalancerPollTimeout, t.jig.PollInterval, t.httpClient, false))
```

Comment:
I'm not sure if I'm understanding what you're asking - I'm also new to this polling package.  Looks like `framework.PollURL` utilizes `wait.PollImmediate` with the passed loadbalancer timeout and interval.

However, this logic is identical to the `test` func in `upgrades/service.go` - it calls `jig.TestReachableHTTP` which utilizes `wait.PollImmediate` with the default `Poll` default of 2 seconds. https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/service_util.go#L668